src/test/java/org/assertj/core/internal/objects/Objects_assertHasToString_Test.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2015 the original author or authors.
+ */
+package org.assertj.core.internal.objects;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.ObjectsBaseTest;
+import org.assertj.core.test.Person;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.assertj.core.error.ShouldHaveSameClass.shouldHaveSameClass;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.test.TestFailures.failBecauseExpectedAssertionErrorWasNotThrown;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Tests for <code>{@link org.assertj.core.internal.Objects#assertHasToString(org.assertj.core.api.AssertionInfo, Object, Object)}</code>.
+ *
+ * @author Nicolas François
+ * @author Joel Costigliola
+ * @author Alexander Bischof
+ */
+public class Objects_assertHasToString_Test extends ObjectsBaseTest {
```

Comment:
this test is no where near the expected quality !!!
3 of 4 tests checks `assertHasSameClassAs` instead of `assertHasToString`
The only test that uses `assertHasToString` is named `should_pass_if_actual_has_same_type_as_other` which is bad.